### PR TITLE
Default secure values for storage accounts

### LINKED ISSUE #770: Set default values to a secure value - Storage Account
All default values should comply with a security baseline, e.g. [NIST 800](https://docs.microsoft.com/en-us/azure/governance/policy/samples/nist-sp-800-53-r5)

The [build-in policies of Azure](https://github.com/Azure/azure-policy) can be used as a reference.

The task would be to scan over each of the following policies and make sure, that the module is per default complying to them.

The following policies are the NIST 800 ones:

\built-in-policies\policyDefinitions\Storage\ASC_Storage_DisallowPublicBlobAccess_Audit.json
\built-in-policies\policyDefinitions\Storage\Classic_AuditForClassicStorages_Audit.json
\built-in-policies\policyDefinitions\Storage\GeoRedundant_StorageAccounts_Audit.json
\built-in-policies\policyDefinitions\Storage\StorageAccountCustomerManagedKeyEnabled_Audit.json
\built-in-policies\policyDefinitions\Storage\StorageAccountInfrastructureEncryptionEnabled_Audit.json
\built-in-policies\policyDefinitions\Storage\StorageAccountOnlyVnetRulesEnabled_Audit.json
\built-in-policies\policyDefinitions\Storage\StorageAccountPrivateEndpointEnabled_Audit.json
\built-in-policies\policyDefinitions\Storage\StorageCache_CMKEnabled.json
\built-in-policies\policyDefinitions\Storage\StorageSync_PrivateEndpoint_AuditIfNotExists.json
\built-in-policies\policyDefinitions\Storage\Storage_AuditForHTTPSEnabled_Audit.json
\built-in-policies\policyDefinitions\Storage\Storage_EncryptionScopesShouldUseCMK_Audit.json
\built-in-policies\policyDefinitions\Storage\Storage_NetworkAcls_Audit.json
\built-in-policies\policyDefinitions\Azure Government\Storage\Storage_EncryptionScopesShouldUseCMK_Audit.json

### PR DESCRIPTION
**Pull request by elanzel**
# Change
- added requireInfrastructureEncryption in deploy.bicep
- Updated parameters descriptions for requireInfrastructureEncryption, privateEndpoints, networkAcls
- set parameters for privateEndpoints, networkAcls, requireInfrastructureEncryption on parameters.json

[Storage: StorageAccounts](https://github.com/Azure/ResourceModules/actions/workflows/ms.storage.storageaccounts.yml)

close #770

## Type of Change
- [X] New feature (non-breaking change which adds functionality)

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (readme)
- [x] I did format my code

### CODE REVIEW

File: arm/Microsoft.Storage/storageAccounts/deploy.bicep [hunk 1]
```diff
@@ -71,7 +74,7 @@ param queueServices object = {}
 @description('Optional. Table service and tables to create.')
 param tableServices object = {}
 
-@description('Optional. Indicates whether public access is enabled for all blobs or containers in the storage account.')
+@description('Optional. Indicates whether public access is enabled for all blobs or containers in the storage account. For security reasons, it is recommended to set it to false.')
 param allowBlobPublicAccess bool = true
```
Comment (MrMCake) [inline arm/Microsoft.Storage/storageAccounts/deploy.bicep:78]: Should we then not also change the default value?
```suggestion
param allowBlobPublicAccess bool = false
```

File: arm/Microsoft.Storage/storageAccounts/readme.md [hunk 2]
```diff
@@ -27,7 +27,8 @@ This module is used to deploy a storage account, with the ability to deploy 1 or
 
 | Parameter Name | Type | Default Value | Possible Values | Description |
 | :-- | :-- | :-- | :-- | :-- |
-| `allowBlobPublicAccess` | bool | `True` |  | Optional. Indicates whether public access is enabled for all blobs or containers in the storage account. |
+| `allowBlobPublicAccess` | bool | `True` |  | Optional. Indicates whether public access is enabled for all blobs or containers in the storage account. For security reasons, it is recommended to set it to false. |
```
Comment (MrMCake) [inline arm/Microsoft.Storage/storageAccounts/readme.md:30]: same here

```suggestion
| `allowBlobPublicAccess` | bool | `false` |  | Optional. Indicates whether public access is enabled for all blobs or containers in the storage account. For security reasons, it is recommended to set it to false. |
```

File: arm/Microsoft.Storage/storageAccounts/deploy.bicep (hunk 1 above)
Comment (elanzel) [reply arm/Microsoft.Storage/storageAccounts/deploy.bicep]: Values in deploy.bicep and descriptions in readme file updated

File: arm/Microsoft.Storage/storageAccounts/readme.md (hunk 2 above)
Comment (elanzel) [reply arm/Microsoft.Storage/storageAccounts/readme.md]: readme file updated

**Review by rahalan — APPROVED** [review]

**Pull request merged**